src/Badge.js

```diff
@@ -25,7 +25,7 @@ class Badge extends React.Component {
 
         return React.cloneElement(element, {
             className: 'mdl-badge',
-            'data-badge': this.props.text
+            'data-badge': this.props.text || undefined
```

Comment:
mdl has a selector that looks like this `.mdl-badge[data-badge]:after` so it renders badge if there's `data-badge` attribute even if it's empty.

And `undefined` does not render attribute at all. Check the DOM in devtools.